collectors/ciscoduo/ciscoduo_collector.js

```diff
@@ -0,0 +1,227 @@
+        let query = this._getDuoQuery(state);
+        
+        client.jsonApiCall('GET', state.duo_endpoint, query, function(res){
+            if (res.stat !== 'OK') {
+                console.error('CDUO000003 API call returned error: ' + res.message);
+                return callback(res.message);
+            } else {
+                return _handleDuoResults(res, callback);
+            }
+        });
+    }
+    
+    _getDuoQuery(state) {
+        let query;
+        
+        switch (state.duo_endpoint) {
+            case DUO_ENDPOINT_AUTH:
+                if (state.next_offset) {
+                    query = {
+                        next_offset: state.next_offset,
+                        limit: LIST_PAGE_SIZE
+                    };
+                } else {
+                    query = {
+                        mintime: state.mintime,
+                        maxtime: state.maxtime,
+                        limit: LIST_PAGE_SIZE
+                    };
+                }
+                break;
+            default:
+                query = {
+                    mintime: state.mintime
+                }
+                break;
+        }
+        return query;
+    }
+    
+    _handleDuoResults(res, callback) {
+        let logs;
+        let metadata;
+        let newState;
+        
+        switch (state.duo_endpoint) {
+            case DUO_ENDPOINT_AUTH:
+                logs = res.response.authlogs;
+                metadata = res.response.metadata;
+            case DUO_ENDPOINT_ADMIN:
+            case DUO_ENDPOINT_TELE:
+            case DUO_ENDPOINT_ENROLL:
+                logs = res.response;
+                // Assumption that the last message timestamp is a mintime for the next collection step
+                metadata = logs[logs.length-1];
+            default:
+                newState = collector._getNextCollectionState(state, metadata);
+                console.info(`CDUO000002 Next collection in ${newState.poll_interval_sec} seconds`);
+                return callback(null, logs, newState, newState.poll_interval_sec);
+        }
+    }
+    
+    _getNextCollectionState(curState, metadata) {
+        // There is an intentional two minute delay in new log availability in the API response.
+        // Let's be safe.
+        const nowMoment = moment().subtract(5, 'minutes');
+        
+        switch (curState.duo_endpoint) {
+            case DUO_ENDPOINT_AUTH:
+                if (metadata.next_offset) {
+                    return {
+                        duo_endpoint: curState.duo_endpoint,
+                        mintime: curState.mintime,
+                        maxtime: curState.maxtime,
+                        poll_interval_sec: 1,
+                        next_offset: metadata.next_offset.join()
+                    };
+                } else {
+                    const curMaxMoment = moment(Number(curState.maxtime));
```

Comment:
I would use `parseInt()` if this is a timestamp. Though I admit in this case that may stray into pedantry.

https://www.freecodecamp.org/forum/t/number-x-vs-parseint-x/188411